pkg/controller/plan/adapter/openstack/client.go

```diff
@@ -943,6 +951,12 @@ func (r *Client) ensureImageUpToDate(vm *libclient.VM, image *libclient.Image, v
 		return
 	}
 
+	if inventoryImage.Status != string(image.Status) {
+		r.Log.Info("image status is not in sync, waiting...",
+			"vm", vm.Name, "image", inventoryImage.Name, "status", inventoryImage.Status)
```

Comment:
don't we need to set `ready` to false?

Reply:
set where?

Reply:
ah the definitions in lines 938-941 are not within a function (why don't we have them at the top?..), so upToDate is necessarily false when we get here..

Reply:
because it's not possible to do type aliasing within a function

Reply:
> because it's not possible to do type aliasing within a function

I meant at the top of the file instead of having them between the functions

Reply:
let's consider doing it separately